docs/reference/search.asciidoc

```diff
@@ -125,9 +125,12 @@ Setting this value to `-1` resets the global search timeout to no timeout.
 [[global-search-cancellation]]
 === Search Cancellation
 
-Searches can be cancelled using standard <<task-cancellation,task cancellation>>
-mechanism. By default, a running search only checks if it is cancelled or
-not on segment boundaries, therefore the cancellation can be delayed by large
+Searches are automatically cancelled when the http connection that they came
+in from gets closed. It is fundamental that the http client sending requests
+closes connections whenever requests time out or are aborted. Automatic
+cancellation relies on standard  <<task-cancellation,task cancellation>>
+mechanism. By default, a running search only checks if it is cancelled or not
```

Comment:
This is unrelated to this change but the default value for low_level_cancellation is now true so the doc is not accurate. We should remove this sentence entirely: `By default, a running search only checks if it is cancelled or not...`.